nav2_amcl/src/amcl_node.cpp

```diff
@@ -1053,31 +1051,12 @@ AmclNode::handleMapMessage(const nav_msgs::msg::OccupancyGrid & msg)
       global_frame_id_.c_str());
   }
 
-  freeMapDependentMemory();
-  // Clear queued laser objects because they hold pointers to the existing
-  // map, #5202.
-  lasers_.clear();
-  lasers_update_.clear();
-  frame_to_laser_.clear();
-
   map_ = convertMap(msg);
 
 #if NEW_UNIFORM_SAMPLING
   createFreeSpaceVector();
 #endif
-  // Create the particle filter
-  initParticleFilter();
-  // resample_count = 0 is extra
-
-  // Instantiate the sensor objects
-  motion_model_.reset();
-  motion_model_ = std::unique_ptr<nav2_amcl::MotionModel>(nav2_amcl::MotionModel::createMotionModel(
```

Comment:
Why was this ever starting the motion model?

Reply:
Originally on every map update, pf was getting reset and all the map data was getting cleared and the motion model was getting deleted and reinitialized.